spec/System/AtomicWrite/Writer/ByteStringSpec.hs

```diff
@@ -134,3 +207,57 @@ spec = do
         resultStat <- getFileStatus filePath
 
         fileMode resultStat `shouldBe` 0o100606
+  describe "atomicWriteFileWithMode'" $ do
```

Comment:
Thanks for adding a good test coverage. Though this file will need to be moved to another spec file, along with the new spec module